src/server/mod.rs

```diff
@@ -473,9 +429,7 @@ impl LanguageServer for LspServer {
                         lsp::TextDocumentSyncKind::FULL,
                     ),
                 ),
-                type_definition_provider: None,
-                workspace: None,
-                workspace_symbol_provider: None,
+                ..Default::default()
```

Comment:
The type itself defines the default, by implementing `Default, and [in this case, it's derived](https://github.com/gluon-lang/lsp-types/blob/9e6b35372685db1d9dc910964fbea756fbc26966/src/lib.rs#L1756). What that means is that the types members all _also_ implement `Default`. The way the lsp spec is written, it's additive, so most of the members are all `Option<T>`; the `Default` of `Option<T>` is `None`. This one line says "take the default" instead of explicitly defining the `None` values for all those attributes.

Originally, I wanted to be explicit here, and list all the capabilities. What that ended up looking like in practice was that it was difficult to really understand what we implemented on purpose and what was "we just put this here to make the compiler happy." This tends to communicate more of our intent than it did before.